src/core/thread/mle_router.cpp

```diff
@@ -3728,12 +3747,32 @@ ThreadError MleRouter::AppendPendingDataset(Message &aMessage)
 {
     ThreadError error = kThreadError_None;
     Tlv tlv;
+    MeshCoP::Dataset *dataset;
```

Comment:
We can avoid duplication of code by defining a new method in `Dataset` that both `AppendActiveDataset()` and `AppendPendingDataset()` can share.